src/table/use-sticky-header.ts

```diff
@@ -60,7 +60,7 @@ export const useStickyHeader = (
         primaryTable.style.marginTop = '';
       }
     };
-  });
+  }, [secondaryTableRef, tableWrapperRef, syncColumnHeaderWidths, items]);
```

Comment:
This is not future-proof, because `items` array can change without changing the actual content (for example, refreshing same data).

The same flickering is still possible under some conditions. What will happen if we remove the return value from this effect completely? Does it cause any new issues?

Reply:
nope, that solves it because the cleanup is what actually causes the flickering
that was my first solution but didn't want to remove the clean up.

Reply:
I cannot see the scenario where not having this cleanup breaks anything. When table unmounts, all of these nodes are destroyed anyway. And during table lifetime `syncColumnHeaderWidths` is always called, as we expect.

Reply:
removed the cleanup and hook dependancies array and the angular hack.
but kept the test in case we attempt to break this again